### PR TITLE
Revisit nginx docs

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#15476

@cokiengchiara please take over. 

@giantswarm/team-halo could you verify if this stuff with the loadbalancers is correct? (not in the changes I've made)

### CODE REVIEW

**Comment by cokiengchiara** [comment]
@ubergesundheit ,

I suggest we have two docs:
1. Running Multiple NGINX Controllers
2. Services of type LoadBalancer

And propose the following change:
- [ ] Change "Services of type LoadBalancer and Multiple Ingress Controllers" -> "Services of type LoadBalancer"
- [ ] Move all (still relevant) instructions on 'how to run multiple NGINX from "Services of type LoadBalancer" article to "Running Multiple NGINX Controllers" article

WDYT? This removes duplicate info. I tried to do the above, but I'm not knowledgeable enough to do it properly 😓  Can you do it? Or we could get on call if you want to work on it together

**Comment by ubergesundheit** [comment]
> WDYT? This removes duplicate info. I tried to do the above, but I'm not knowledgeable enough to do it properly sweat Can you do it? Or we could get on call if you want to work on it together

Good Idea! I'll try to untangle them. Would be cool if you would check for sanity etc afterwards

**Comment by ubergesundheit** [comment]
ping @cokiengchiara please check the latest changes

**Comment by marians** [comment]
Heads-up: https://github.com/giantswarm/docs/pull/778 is going to be merged soon. The guide you are editing is moving to `/advanced/ingress/multi-nginx-ic/`

See https://docs-staging.c68pn.k8s.gollum.westeurope.azure.gigantic.io/advanced/ingress/multi-nginx-ic/ for a staging version.

File: src/content/guides/multi-nginx/index.md [hunk 1]
```diff
@@ -5,20 +5,56 @@ type: page
...
+
+Not specifying the annotation will lead to multiple ingress controllers claiming the same ingress. Specifying a value which does not match the class of any existing ingress controllers will result in all ingress controllers ignoring the ingress.
+
+Further note that if you are running additional Ingress Controllers you might need to configure them so their Ingress Class does not collide with the class of our default NGINX Ingress Controller. For the community supported NGINX Ingress Controller this is described in the [official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/).
+
+Most NGINX configuration options have NGINX-wide defaults. They can also be overriden on a per-Ingress resource level.
+
+In each case below, one installs a second NGINX with a different NGINX global-only configuration. Ingress resources managed by this NGINX installation cannot be customized on a per-Ingress resource level.
+
+Further information on configuring NGINX Ingress Controller can be found in the [Services of type LoadBalancer]
```
Comment (cokiengchiara) [inline src/content/guides/multi-nginx/index.md:48]:
```suggestion
Further information on configuring NGINX Ingress Controller can be found in the article, [Services of type LoadBalancer]({{< relref "/content/guides/services-of-type-loadbalancer/index.md" >}}).
```

File: src/content/guides/multi-nginx/index.md [hunk 2]
```diff
@@ -5,20 +5,56 @@ type: page
...
+
+or
+
+```yaml
+kubernetes.io/ingress.class: "nginx-internal"
+```
+
+Not specifying the annotation will lead to multiple ingress controllers claiming the same ingress. Specifying a value which does not match the class of any existing ingress controllers will result in all ingress controllers ignoring the ingress.
+
+Further note that if you are running additional Ingress Controllers you might need to configure them so their Ingress Class does not collide with the class of our default NGINX Ingress Controller. For the community supported NGINX Ingress Controller this is described in the [official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/).
```
Comment (cokiengchiara) [inline src/content/guides/multi-nginx/index.md:42]: We no longer have default (preinstalled) nginx in most clusters https://docs.giantswarm.io/general/releases/#apps - how does that change this?

Maybe sth like this?
```suggestion
Additionally, please ensure the Ingress Class of each of your Ingress Controllers do not collide with each other. For the community supported NGINX Ingress Controller this is described in the [official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/multiple-ingress/).
```

**Comment by cokiengchiara** [comment]
@ubergesundheit looking good. Thanks! One final suggestion on the order of things.

Would it make it clearer if
1. We added tl;dr at the top like below.
2. Rearrange sections like below.

---

tl;dr To install more than 1 NGINX Ingress Controller:
1. Install a second NGINX IC app (and subsequent apps) with a different NGINX global-only configuration. Ingress resources managed by this NGINX installation cannot be customized on a per-Ingress resource level.
2. Annotate each Ingress with the appropriate class. Make sure the Ingress Class of each Ingress Controller do not collide with each other

## Step 1: Install a second NGINX Ingress Controller

### Separating public from internal ingress traffic
### Using weak ciphers for legacy clients

## Step 2: Annotate each Ingress with the appropriate class

**Comment by ubergesundheit** [comment]
@cokiengchiara wasn't sure how to apply your suggestion. please take a look

File: src/content/advanced/ingress/multi-nginx-ic/index.md [hunk 3]
```diff
@@ -10,20 +10,63 @@ aliases:
...
+NGINX ingress controller handles [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resources, routing traffic from outside the Kubernetes cluster to services within the cluster.
 
-Most of the NGINX configuration options have NGINX wide defaults but can also be overriden on per Ingress resource level. In each of the covered multi-NGINX use cases, one essentially installs second NGINX with different NGINX global-only configuration, i.e. configuration that applies to all Ingress resources managed by that NGINX installation which cannot be customized on per-Ingress resource level.
+Starting with [NGINX IC v1.8.0](https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/CHANGELOG.md#180---2020-07-24), one can install multiple NGINX ingress controllers in a Kubernetes cluster. The optional NGINX Ingress Controller can be [installed as as an App on your cluster]({{< relref "/content/getting-started/ingress-controller/index.md" >}}). This Ingress Controller is registered with the default `nginx` Ingress Class.
```
Comment (cokiengchiara) [inline src/content/advanced/ingress/multi-nginx-ic/index.md:25]:
```suggestion
Starting with [NGINX IC v1.8.0](https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/CHANGELOG.md#180---2020-07-24), one can install multiple NGINX ingress controllers in a Kubernetes cluster. The optional NGINX Ingress Controller can be [installed as as an App on your cluster]({{< relref "/content/getting-started/ingress-controller/index.md" >}}). This Ingress Controller is registered with the default `nginx` Ingress Class.
```
```suggestion
Starting with [NGINX IC v1.8.0](https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/CHANGELOG.md#180---2020-07-24), one can install multiple NGINX ingress controllers in a Kubernetes cluster. The optional NGINX Ingress Controller can be [installed as an App on your cluster]({{< relref "/content/getting-started/ingress-controller/index.md" >}}). This Ingress Controller is registered with the default `nginx` Ingress Class.
```

Comment (cokiengchiara) [inline src/content/advanced/ingress/multi-nginx-ic/index.md:25]:
```suggestion
Starting with [NGINX IC v1.8.0](https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/CHANGELOG.md#180---2020-07-24), one can install multiple NGINX ingress controllers in a Kubernetes cluster. The optional NGINX Ingress Controller can be [installed as an App on your cluster]({{< relref "/content/getting-started/ingress-controller/index.md" >}}). This Ingress Controller is registered with the default `nginx` Ingress Class.
```

File: src/content/advanced/ingress/service-type-loadbalancer/index.md [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
 aliases:
   - /guides/services-of-type-loadbalancer-and-multiple-ingress-controllers/
+  - /advanced/ingress/service-type-loadbalancer-multi-ic/
 owner:
   - https://github.com/orgs/giantswarm/teams/team-halo
+user_questions:
+  - How can I expose Services to the internet?
+  - How do I configure an Ingress Controller that is behind an internal ELB for traffic between services within the VPC (or a group of peered VPCs)?
+  - How do I configure an Ingress Controller behind an ELB that already terminates SSL?
+  - How do I configure an Ingress Controller with different functionality or performance?
```
Comment (cokiengchiara) [inline src/content/advanced/ingress/service-type-loadbalancer/index.md:17]:
```suggestion
```
Moving this to the multi nginx doc

File: src/content/advanced/ingress/multi-nginx-ic/index.md [hunk 5]
```diff
@@ -10,20 +10,63 @@ aliases:
   - /guides/multi-nginx/
 owner:
   - https://github.com/orgs/giantswarm/teams/team-halo
+user_questions:
+  - How do I install multiple NGINX Ingress Controllers?
+  - How do I separate internal and external Services?
+  - How do I configure NGINX Ingress Controller for internal traffic?
+  - How do I override the NodePorts on KVM Ingresses?
+  - How do I configure NGINX Ingress Controller to allow weak ciphers for legacy devices?
```
Comment (cokiengchiara) [inline src/content/advanced/ingress/multi-nginx-ic/index.md:18]:
```suggestion
  - How do I configure NGINX Ingress Controller to allow weak ciphers for legacy devices?
  - How do I configure an Ingress Controller with different functionality or performance?
```

**Review by cokiengchiara — CHANGES_REQUESTED** [review]
LGTM, thanks so much @ubergesundheit !

* Checked that the content is not duplicated in the two docs
* General clarity and flow
* User questions (thanks so much Gerald!)

I couldn't check for technical accuracy but I assume Gerald took care of that.

**Comment by cokiengchiara** [comment]
> Heads-up: #778 is going to be merged soon. The guide you are editing is moving to `/advanced/ingress/multi-nginx-ic/`

@ubergesundheit LGTM. I made few more minor comments. Otherwise, feel free to merge. Also I don't know the implication of what Marian is saying, but I'm sure you do 😀

**Comment by ubergesundheit** [comment]
@cokiengchiara giev approve plz

**Review by cokiengchiara — APPROVED** [review]
